synchronicity/synchronizer.py

```diff
@@ -644,7 +650,7 @@ def _wrap_class(self, cls, interface, name, target_module=None):
         if name is None:
             name = _CLASS_PREFIXES[interface] + cls.__name__
 
-        new_cls = type.__new__(type, name, bases, new_dict)
+        new_cls = types.new_class(name, bases, exec_body=lambda ns: ns.update(new_dict))
```

Comment:
TIL about `types.new_class` which is apparently the new proper way of runtime-creating new classes. `type.__new__` seems to have some issues with newer additions to the language, like `__orig_bases__` support